server-core/engine/src/main/java/io/infectnet/server/engine/core/world/strategy/AStarPathFinderStrategy.java

```diff
@@ -0,0 +1,298 @@
+package io.infectnet.server.engine.core.world.strategy;
+
+import io.infectnet.server.engine.core.world.Position;
+import io.infectnet.server.engine.core.world.Tile;
+import io.infectnet.server.engine.core.world.World;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import static io.infectnet.server.engine.core.world.TileType.CAVE;
+
+/**
+ * Class for Path finding in the world between two Positions, using the A* algorithm.
+ */
+public class AStarPathFinderStrategy implements PathFinderStrategy{
+
+    /**
+     * A heuristic to use with value calculating.
+     */
+    private final Heuristic heuristic;
+
+    /**
+     * The world, in which we want to find the path.
+     */
+    private World world;
```

Comment:
All fields except the `heuristic` should be removed because they do not permit the usage of the strategy in a multi-threaded environment. Although it's highly unlikely to operate on multiple `World` instances, even that field should be removed too.